R/arc-select.R

```diff
@@ -454,10 +450,91 @@ count_results <- function(req, query) {
   resp <- httr2::resp_body_string(
     httr2::req_perform(
       httr2::req_url_query(n_req, f = "json"),
-      error_call = rlang::caller_env()
+      error_call = error_call
     )
   )
 
   RcppSimdJson::fparse(resp)[["count"]]
 }
 
+#' Set page size and check page size validity
+#'
+#' @noRd
+set_page_size <- function(
+    page_size = NULL,
+    max_records = 25000L,
+    error_call = rlang::caller_env()
+    ) {
+  # if its null, just use max records (default)
+  if (is.null(page_size)) {
+    return(max_records)
+  }
+
+  # ensure its an integer.
+  page_size <- as.integer(page_size)
+  page_size_len <- length(page_size)
+
+  if (page_size_len > 1) {
+    cli::cli_abort(
+      "{.arg page_size} must be length 1 not {page_size_len}",
+      call = error_call
+    )
+  }
+
+  if (page_size < 1) {
+    cli::cli_abort(
+      "{.arg page_size} must be a positive integer.",
+      call = error_call
+    )
+  }
+
+  if (page_size > max_records) {
+    cli::cli_abort(
+      "{.arg page_size} ({page_size}) cannot excede layer's {.field maxRecordCount} property ({max_records})",
```

Comment:
hmmmm. I agree with this check. _But_ the presence of a default value for `max_records` might mean that if ~~we are~~ I am lazy we might report the wrong number (25000) instead of the layers value.

Reply:
The default value should not be used as long as the layer has a maxRecordsCount value but I dropped it regardless.